python/lsst/ts/MTAirCompressor/aircompressor_model.py

```diff
@@ -0,0 +1,163 @@
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ModbusError", "MTAirCompressorModel"]
+
+import enum
+
+# although pymodbus supports asyncio, it's uselless to use asyncio version
+# as there isn't any extra processing which can occur while waiting for modbus
+# data
+from pymodbus.client.sync import ModbusTcpClient as ModbusClient
+import pymodbus.exceptions
+
+
+# Address of registers of interest. Please see Delcos XL documentation for details
+class Registers(enum.IntEnum):
+    # telemetry block
+    WATER_LEVEL = 0x1E
+
+    TARGET_SPEED = 0x22
+    MOTOR_CURRENT = 0x23
+    HEATSINK_TEMP = 0x24
+    DCLINK_VOLTAGE = 0x25
+    MOTOR_SPEED_PERCENTAGE = 0x26
+    MOTOR_SPEED_RPM = 0x27
+    MOTOR_INPUT = 0x28
+    COMPRESSOR_POWER_CONSUMATION = 0x29
+    COMPRESSOR_VOLUME_PERCENTAGE = 0x2A
+    COMPRESSOR_VOLUME = 0x2B
+    GROUP_VOLUME = 0x2C
+    STAGE_1_OUTPUT_PRESSURE = 0x2D
+    LINE_PRESSURE = 0x2E
+    STAGE_1_OUTPUT_TEMPERATURE = 0x2F
+
+    RUNNIG_HOURS = 0x39  # 64bit, 2 registers
+    LOADED_HOURS = 0x3B  # 64 bit, 2 registers
+    LOWEST_SERVICE_COUNTER = 0x3C
+    RUN_ON_TIMER = 0x3D
+    LOADED_HOURS_50_PERECENT = 0x3E  # 64 bit, 2 registers
+
+    STATUS = 0x30  # flags - started, ..
+    ERROR_E400 = 0x63  # 16 registers with error and warning flags
+
+    SOFTWARE_VERSION = 0xC7  # 14 ASCII registers
+    SERIAL_NUMER = 0xD5  # 9 ASCII registers
+
+    REMOTE_CMD = 0x12B  # power on/off, if remote commanding is enabled
+    RESET = 0x12D  # reset errors & warnings
+
+
+class ModbusError(RuntimeError):
+    """Exception raised on modbus errors. Please note that shall be superset by
+    pymodbus solution, if it ever materialize, See:
+    https://github.com/riptideio/pymodbus/issues/298
+
+    Parameters
+    ----------
+    modbusException : `pymodbus.exception.ModbusException`
+        ModBus orignal exception
+    message : `str`
+        Message associated with the error
+    """
+
+    def __init__(self, modbusException, message=""):
+        self.exception = modbusException
+        self.message = message
+
+
+class MTAirCompressorModel(ModbusClient):
+    """Model for compressor. Handles compressor communication. Throws
```

Comment:
Thank you for the clear description. Please do reformat to meet our usual standards, e.g.:
* Doc strings should start with a single sentence (preferably that fits on one line, though that isn't always possible). Then a blank line, then any additional description (if needed).
* Constructor arguments should be documented in a Parameters section in this the class doc string (this one)